tests/initialization_tests.hpp

```diff
@@ -55,4 +55,79 @@ TEST(MatrixInitTest, CreatesMatrixFromCSV) {
     EXPECT_EQ(mat, test_with);
 }
 
+TEST(MatrixInitTest, CreatesMatrixObjectFromString1DVector) {
+    std::vector<std::string> vec;
+    Matrix mat;
+
+    EXPECT_EQ(typeid(mat).name(), typeid(matrix.init(vec)).name());
+}
+
+TEST(MatrixInitTest, CreatesMatrixObjectFromDouble1DVector) {
+    std::vector<double> vec;
+    Matrix mat;
+
+    EXPECT_EQ(typeid(mat).name(), typeid(matrix.init(vec)).name());
+}
+
+TEST(MatrixInitTest, CreatesMatrixObjectFromString) {
+    Matrix mat;
+
+    EXPECT_EQ(typeid(mat).name(), typeid(matrix.init("")).name());
+}
+
+TEST(MatrixInitTest, CreatesMatrixObjectFromDouble) {
+    Matrix mat;
+
+    EXPECT_EQ(typeid(mat).name(), typeid(matrix.init(0)).name());
+}
+
+TEST(MatrixInitTest, InitializesMatrixFromString1DVector) {
+    std::vector<std::string> vec;
+    vec.push_back("1");
+    vec.push_back("2");
+    vec.push_back("3");
+
+    Matrix mat = matrix.init(vec);
+
+    Matrix test_with = matrix.genfromtxt('test_dataset.csv', ',');
```

Comment:
At lines 92, 106, 117 and 127, 'test_dataset.csv' should be "test_dataset.csv". Just correct that and then I will merge this PR.

I have checked everything else and it works fine.